system/lib/wasmfs/proxied_file_backend.cpp

```diff
@@ -84,7 +84,12 @@ class ProxiedBackend : public Backend {
   emscripten::SyncToAsync proxy;
 
 public:
-  ProxiedBackend(backend_t backend) : backend(backend) {}
+  ProxiedBackend(make_backend_t make_backend, void* arg) {
```

Comment:
Maybe we can take just a `std::function<backend_t()>` here and let the C wrapper handle the conversion with a small lambda?